### PR TITLE
Bug 1742172 Support override_attributes metadata

### PR DESCRIPTION
**Pull request by jklukas**
See https://bugzilla.mozilla.org/show_bug.cgi?id=1742172

See https://github.com/mozilla-services/mozilla-pipeline-schemas/pull/709 for the coordinated change to schemas.

### CODE REVIEW

**Comment by jklukas** [comment]
cc @hackebrot since we discussed these tests recently, and this PR rolls in some improvements.

File: ingestion-beam/src/test/java/com/mozilla/telemetry/decoder/SanitizeAttributesTest.java [hunk 1]
```diff
@@ -17,42 +18,82 @@
...
 
   @Test
   public void testSubmissionTimestampGranularity() {
     // Depends on custom doctype injected in bin/download-schemas
-    Map<String, String> attributes = ImmutableMap.<String, String>builder()
+    Map<String, String> inputAttributes = ImmutableMap.<String, String>builder()
         .put(Attribute.DOCUMENT_NAMESPACE, "test") //
         .put(Attribute.DOCUMENT_TYPE, "seconds") //
         .put(Attribute.DOCUMENT_VERSION, "1") //
-        .put("submission_timestamp", "2021-01-01T12:00:00.123456Z") //
```
Comment (whd) [inline ingestion-beam/src/test/java/com/mozilla/telemetry/decoder/SanitizeAttributesTest.java:48]: I don't have enough familiarity with the code to understand why this wasn't an `Attribute` before but it looks uniform now.

**Review by whd — APPROVED** [review]
LGTM from the main logic and test, the JSON deserializing `Builder` stuff :shrug:

**Review by relud — APPROVED** [review]

File: ingestion-beam/src/test/java/com/mozilla/telemetry/decoder/SanitizeAttributesTest.java (hunk 1 above)
Comment (jklukas) [reply ingestion-beam/src/test/java/com/mozilla/telemetry/decoder/SanitizeAttributesTest.java:48]: This is just cleaning up an oversight. That value of the constant is equal to `"submission_timestamp"`.

**Pull request merged**